File: ja-muslims-directory/backend/app/models/restaurant.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey, DateTime, Enum, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.base import Base
import enum


class CuisineType(str, enum.Enum):
    JAMAICAN = "jamaican"
    MIDDLE_EASTERN = "middle_eastern"
    INDIAN = "indian"
    PAKISTANI = "pakistani"
    MEDITERRANEAN = "mediterranean"
    AFRICAN = "african"
    AMERICAN = "american"
    CHINESE = "chinese"
    VEGETARIAN = "vegetarian"
    VEGAN = "vegan"
    SEAFOOD = "seafood"
    FUSION = "fusion"
    OTHER = "other"


class Restaurant(Base):
    __tablename__ = "restaurants"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    address = Column(String, nullable=False)
    parish = Column(String, nullable=False)
    phone = Column(String)
    email = Column(String)
    website = Column(String)
    
    # Restaurant type
    is_halal_certified = Column(Boolean, default=False)
    has_halal_options = Column(Boolean, default=False)
    has_vegetarian_options = Column(Boolean, default=False)
    has_vegan_options = Column(Boolean, default=False)
    
    # Details
    cuisine_types = Column(String)  # Comma-separated list
    opening_hours = Column(Text)
    description = Column(Text)
    
    # Link to business if Muslim-owned
    business_id = Column(Integer, ForeignKey("businesses.id"), nullable=True)
    business = relationship("Business", back_populates="restaurant")
    
    # Menu files
    menu_files = relationship("RestaurantMenu", back_populates="restaurant", cascade="all, delete-orphan")
    
    # Timestamps
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())


class RestaurantMenu(Base):
    __tablename__ = "restaurant_menus"
    
    id = Column(Integer, primary_key=True, index=True)
    restaurant_id = Column(Integer, ForeignKey("restaurants.id"), nullable=False)
    file_name = Column(String, nullable=False)
    file_path = Column(String, nullable=False)
    file_type = Column(String, nullable=False)  # 'pdf' or 'image'
    uploaded_at = Column(DateTime(timezone=True), server_default=func.now())
    
    restaurant = relationship("Restaurant", back_populates="menu_files")

File: ja-muslims-directory/frontend/types/restaurant.ts
export interface RestaurantMenu {
  id: number;
  restaurant_id: number;
  file_name: string;
  file_path: string;
  file_type: 'pdf' | 'image';
  uploaded_at: string;
}

export interface Restaurant {
  id: number;
  name: string;
  address: string;
  parish: string;
  phone?: string;
  email?: string;
  website?: string;
  is_halal_certified: boolean;
  has_halal_options: boolean;
  has_vegetarian_options: boolean;
  has_vegan_options: boolean;
  cuisine_types?: string;
  opening_hours?: string;
  description?: string;
  business_id?: number;
  created_at: string;
  updated_at?: string;
  menu_files: RestaurantMenu[];
}

export interface RestaurantWithBusiness extends Restaurant {
  business_name?: string;
  owner_name?: string;
}

export interface RestaurantFormData {
  name: string;
  address: string;
  parish: string;
  phone?: string;
  email?: string;
  website?: string;
  is_halal_certified: boolean;
  has_halal_options: boolean;
  has_vegetarian_options: boolean;
  has_vegan_options: boolean;
  cuisine_types?: string;
  opening_hours?: string;
  description?: string;
  business_id?: number;
}

export const CUISINE_TYPES = [
  { value: 'jamaican', label: 'Jamaican' },
  { value: 'middle_eastern', label: 'Middle Eastern' },
  { value: 'indian', label: 'Indian' },
  { value: 'pakistani', label: 'Pakistani' },
  { value: 'mediterranean', label: 'Mediterranean' },
  { value: 'african', label: 'African' },
  { value: 'american', label: 'American' },
  { value: 'chinese', label: 'Chinese' },
  { value: 'vegetarian', label: 'Vegetarian' },
  { value: 'vegan', label: 'Vegan' },
  { value: 'seafood', label: 'Seafood' },
  { value: 'fusion', label: 'Fusion' },
  { value: 'other', label: 'Other' },
];

export const PARISHES = [
  'Kingston',
  'St. Andrew',
  'St. Catherine',
  'Clarendon',
  'Manchester',
  'St. Elizabeth',
  'Westmoreland',
  'Hanover',
  'St. James',
  'Trelawny',
  'St. Ann',
  'St. Mary',
  'Portland',
  'St. Thomas',
];